app/start.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# Carnot App - Startup Script
+# Starts both the backend and frontend servers
+
+echo "🚀 Starting Carnot Application..."
+echo ""
+
+# Function to cleanup background processes on exit
+cleanup() {
+    echo ""
+    echo "🛑 Shutting down servers..."
+    kill $BACKEND_PID $FRONTEND_PID 2>/dev/null
+    exit
+}
+
+trap cleanup EXIT INT TERM
+
+# Get absolute path to script directory
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Check if virtual environment exists (either in backend or parent directory)
+if [ -d "$SCRIPT_DIR/backend/venv" ]; then
+    VENV_PATH="$SCRIPT_DIR/backend/venv"
+elif [ -d "$SCRIPT_DIR/../.venv312" ]; then
+    VENV_PATH="$SCRIPT_DIR/../.venv312"
+elif [ -d "$SCRIPT_DIR/../.venv311" ]; then
+    VENV_PATH="$SCRIPT_DIR/../.venv311"
+else
+    echo "❌ Backend virtual environment not found!"
+    echo "Please activate your Python virtual environment and run: pip install -r backend/requirements.txt"
+    exit 1
+fi
```

Comment:
While I like this sanity check, I think the implicit dependency on having `$SCRIPT_DIR/backend/venv` is not guaranteed to be true -- I'd be okay with removing this for now and trusting the person / deployment pipeline invoking the script to set the environment properly.

Perhaps there is some other check we could perform though that would be universal?
For example, that `pip freeze | grep carnot` should return something?